example_app/request_accepts_resource_test.ts

```diff
@@ -0,0 +1,25 @@
+import members from "../tests/members.ts";
+members.test("RequestAcceptsResource", async () => {
+  let response;
+  let json;
+
+  // Accepts the correct type the resource will give
+  response = await members.fetch.get("http://localhost:1667/request-accepts", {
+    headers: {
+      "Accept": "application/json",
+    },
+  });
+  json = JSON.parse(await response.json());
+  members.assert.equals(json['success'], true);
+  members.assert.equals(json.message, "Responding with the accepted content type");
+
+  // Does not accept the type the resource expects
+  response = await members.fetch.get("http://localhost:1667/request-accepts", {
+    headers: {
+      "Accept": "text/html",
+    },
+  });
+  json = JSON.parse(await response.json());
+  await members.assert.equals(json.success, false);
+  await members.assert.equals(json.message, 'This resource only accepts JSON');
+});
```

Comment:
Maybe tests for preference `Accept` values with 3+ possible `Content-Type`, making sure it follows the client precedence?

Reply:
Good shout, working on reworking the tests to be more rigorous